pkg/tsdb/models.go

```diff
@@ -85,3 +88,88 @@ func NewTimeSeries(name string, points TimeSeriesPoints) *TimeSeries {
 		Points: points,
 	}
 }
+
+type DataFrames interface {
+	Encoded() ([][]byte, error)
+	Decoded() (data.Frames, error)
+}
+
+type decodedDataFrames struct {
```

Comment:
I think this can be the same struct declaration as encoded frame, but with a different name if you want a different type.